src/software/simulated_tests/ball_moves_in_direction.py

```diff
@@ -0,0 +1,124 @@
+import software.python_bindings as tbots
+from proto.import_all_protos import *
+
+from software.simulated_tests.validation import (
+    Validation,
+    create_validation_geometry,
+    create_validation_types,
+)
+
+
+class BallMoves(Validation):
+    """Checks if ball is moving forward or backward, i.e. in the +x direction"""
+
+    def __init__(self, initial_ball_position, direction):
+        """
+        Constructs the validation
+        :param initial_ball_position: the initial position of the ball
+        :param direction: True if checking for forward, false if backward
+        """
+        self.last_ball_position = initial_ball_position
+        self.direction = direction
+
+    def get_validation_status(self, world) -> ValidationStatus:
+        """Checks if ball is moving forward or backward
+
+        :param world: The world msg to validate
+        :returns: FAILING if ball doesn't move in the direction
+                  PASSING if ball moves in the direction
+        """
+        validation_status = ValidationStatus.FAILING
+        current_ball_position = tbots.createPoint(
+            world.ball.current_state.global_position
+        )
+
+        if (
+            self.direction and current_ball_position.x() >= self.last_ball_position.x()
```

Comment:
should this check for the y direction as well? or is this what you refer to buy forward and back? In that case then this class should be called `BallMovesXAxis`

Reply:
yes this just focuses on the x-axis. i'll change the name